### ISSUE TITLE
In `compute` clause, the key value should be available but is null

### ISSUE THREAD

**Issue by julianhyde** [julianhyde/morel#67]
In the `compute` clause of `group`, the key value should be available but is null. For example,
```
from (x, y) in [("a", "p"), ("b", "q"), ("a", "r"), ("b", "s")]
  group x compute z = (fn ys => String.concat (x :: ys)) of y
```
references `x` and finds that its value is `null` at runtime, hence throws
```
java.lang.NullPointerException
 at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:770)
 at com.google.common.collect.ImmutableCollection$ArrayBasedBuilder.add(ImmutableCollection.java:477)
 at com.google.common.collect.ImmutableList$Builder.add(ImmutableList.java:758)
 at net.hydromatic.morel.eval.Codes$22.apply(Codes.java:328)
 at net.hydromatic.morel.eval.Codes$ApplyCode.eval(Codes.java:2461)
``` 

A similar query that does not reference `x` succeeds:
```
- from (x, y) in [("a", "p"), ("b", "q"), ("a", "r"), ("b", "s")]
=   group x compute z = (fn ys => String.concat ys) of y;
val it = [{x="a",z="pr"},{x="b",z="qs"}] : {x:string, z:string} list
```

**Comment by julianhyde** [comment]
Separately, it's not entirely clear what variables should be available in the `compute` clause:
* after `group x` clearly `x` is available;
* after `group (x, y)` clearly `x` and `y` are available;
* but after `group x + y` neither `x` nor `y` should be available, and there's no obvious way to reference `x + y`.

**Comment by julianhyde** [comment]
Fixed in 28690e655ed56d98b60d62bbc1af55e2cbb4106c.
